src/java/org/jivesoftware/openfire/plugin/session/RemoteClientSession.java

```diff
@@ -157,20 +165,32 @@ public int incrementConflictCount() {
         return result == null ? 0 : (Integer) result;
     }
 
+/*
+*Changed for OF-1868 
+*/
     @Override
     public boolean isMessageCarbonsEnabled() {
-        return messageCarbonsEnabled;
+        //get the flag from clustered session cache instead of local variable
+        Cache<String,ClientSessionInfo> cache = SessionManager.getInstance().getSessionInfoCache();
+	ClientSessionInfo sessionInfo = cache.get(getAddress().toString());
+	return sessionInfo != null && sessionInfo.isMessageCarbonsEnabled();
     }
 
+/*
+*Changed for OF-1868 
+*/
     @Override
     public void setMessageCarbonsEnabled(boolean enabled) {
-        this.messageCarbonsEnabled = enabled;
+        doClusterTask(new SetMessageCarbonsTask(address, enabled)); //cluster the flag to other member ndoes
     }
 
+/*
+*Changed for OF-1868 
+*/
     @Override
     public boolean hasRequestedBlocklist()
     {
-        // After it's determined that a session has requested a blocklist, this value will never revert back to false.
+        /*// After it's determined that a session has requested a blocklist, this value will never revert back to false.
```

Comment:
Could you remove the additonal end text that looks like and end comment, but isn't?

Reply:
i removed the whole code ;)